upload-api/tables/consumer.js

```diff
@@ -41,13 +42,91 @@ export class ConflictError extends Failure {
   }
 }
 
+/**
+ * 
+ * @param {DynamoDBClient} dynamoDb
+ * @param {string} tableName 
+ * @param {import('@ucanto/interface').DID} consumer 
+ * @returns {Promise<import('@web3-storage/upload-api').ProviderDID[]>}
+ */
+async function getStorageProviders (dynamoDb, tableName, consumer) {
+  const cmd = new QueryCommand({
+    TableName: tableName,
+    IndexName: 'consumer',
+    KeyConditions: {
+      consumer: {
+        ComparisonOperator: 'EQ',
+        AttributeValueList: [{ S: consumer }]
+      }
+    },
+    AttributesToGet: ['provider']
+  })
+  const response = await dynamoDb.send(cmd)
+  return response.Items ? response.Items.map(item => {
+    const row = unmarshall(item)
+    return /** @type {import('@web3-storage/upload-api').ProviderDID} */ (row.provider)
+  }) : []
+}
+
```

Comment:
any plan to handle pagination? Probably future problem, but maybe we should comment it and create an issue

Reply:
ah right yea not handling that - will comment and file an issue, it probably won't be a problem for a long time (if ever?) but worth tracking

Reply:
huh apparently Query will return 1MB of results? https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Query.Pagination.html

I can't imagine this ever being larger than 1MB, so maybe we just don't need to worry about it?

Reply:
I would add a comment in the code just in case